File: crossecore-java/src/java_/ArrayCreation.java
package java_;
public interface ArrayCreation 
 extends Expression
{
	
	public static com.crossecore.ocl.QuickSet<ArrayCreation> allInstances_ = new com.crossecore.ocl.QuickSet<ArrayCreation>(ArrayCreation.class);
	
		
	public static com.crossecore.ocl.QuickSet<ArrayCreation> allInstances(){
		
		com.crossecore.ocl.QuickSet<ArrayCreation> result = new com.crossecore.ocl.QuickSet<ArrayCreation>(ArrayCreation.class);
		result.addAll(ArrayCreation.allInstances_);
		
		
		return result;
	}
	
	//public static com.crossecore.ocl.QuickSet<ArrayCreation> allInstances = new com.crossecore.ocl.QuickSet<ArrayCreation>(ArrayCreation.class);
	
	com.crossecore.ocl.OrderedSet<Expression> getDimensions();
	
	ArrayInitializer getInitializer();
	void setInitializer(ArrayInitializer value);
	
	TypeAccess getType();
	void setType(TypeAccess value);
	
}

File: crossecore-typescript/java_/ArrayCreation.ts
import {OrderedSet} from "ecore/OrderedSet";
import {Set} from "ecore/Set";
import {Expression} from "./Expression";
import {TypeAccess} from "./TypeAccess";
import {ArrayInitializer} from "./ArrayInitializer";

export interface ArrayCreation
extends Expression

{
	
	dimensions: OrderedSet<Expression>;
	initializer:ArrayInitializer;
	type:TypeAccess;
	

}

